src/lib/common/sem.cpp

```diff
@@ -459,13 +459,131 @@ int timeStatSemGive(const char* who, const char* what)
 
 
 /* ****************************************************************************
+*
+* reqSemGet - 
+*/
+const char* reqSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&reqSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
+*
+* transSemGet - 
+*/
+const char* transSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&transSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
+*
+* cacheSemGet - 
+*/
+const char* cacheSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&cacheSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
+*
+* timeStatSemGet - 
+*/
+const char* timeStatSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&timeStatSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
 *  curl context
 */
 
+static bool contexts_mutex_taken   = false;
+static int  contexts_mutex_errors  = 0;
+static bool contexts_mutex2_taken  = false;
```

Comment:
I believe we agreed to use the name endpoint_mutexes, especially note that there is not only **one** mutex, but one per **endpoint**.